datafusion/substrait/src/logical_plan/consumer/expr/scalar_function.rs

```diff
@@ -254,3 +284,86 @@ impl BuiltinExprBuilder {
         }))
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::arg_list_to_binary_op_tree;
+    use crate::extensions::Extensions;
+    use crate::logical_plan::consumer::tests::TEST_SESSION_STATE;
+    use crate::logical_plan::consumer::{DefaultSubstraitConsumer, SubstraitConsumer};
+    use datafusion::arrow::datatypes::{DataType, Field, Schema};
+    use datafusion::common::{DFSchema, Result, ScalarValue};
+    use datafusion::logical_expr::{Expr, Operator};
+    use insta::assert_snapshot;
+    use substrait::proto::expression::literal::LiteralType;
+    use substrait::proto::expression::{Literal, RexType, ScalarFunction};
+    use substrait::proto::function_argument::ArgType;
+    use substrait::proto::{Expression, FunctionArgument};
+
+    /// Test that large argument lists for binary operations do not crash the consumer
+    #[tokio::test]
+    async fn test_binary_op_large_argument_list() -> Result<()> {
+        // Build substrait extensions (we are using only one function)
+        let mut extensions = Extensions::default();
+        extensions.functions.insert(0, String::from("or:bool_bool"));
+        // Build substrait consumer
+        let consumer = DefaultSubstraitConsumer::new(&extensions, &TEST_SESSION_STATE);
+
+        // Build arguments for the function call, this is basically an OR(true, true, ..., true)
+        let arg = FunctionArgument {
+            arg_type: Some(ArgType::Value(Expression {
+                rex_type: Some(RexType::Literal(Literal {
+                    nullable: false,
+                    type_variation_reference: 0,
+                    literal_type: Some(LiteralType::Boolean(true)),
+                })),
+            })),
+        };
+        let arguments = vec![arg; 50000];
+        let func = ScalarFunction {
+            function_reference: 0,
+            arguments,
+            ..Default::default()
+        };
+        // Trivial input schema
+        let schema = Schema::new(vec![Field::new("a", DataType::Boolean, false)]);
+        let df_schema = DFSchema::try_from(schema).unwrap();
+
+        // Consume the expression and ensure we don't crash
+        let _ = consumer.consume_scalar_function(&func, &df_schema).await?;
+        Ok(())
+    }
+
+    fn n(n: i64) -> Expr {
+        Expr::Literal(ScalarValue::Int64(Some(n)))
+    }
+
+    #[test]
+    fn arg_list_to_binary_op_tree_1_arg() -> Result<()> {
+        let expr = arg_list_to_binary_op_tree(Operator::Or, vec![n(1)])?;
```

Comment:
you can use `lit(1)` directly instead of n(1)

Reply:
Makes sense, I switched to a more explicit function name (`int64_literals`) taking a slice of integers to keep a compact code.